eclipse-theia/theia
# update electron to version 11

update electron-store to version 7. Its required by electron 11.
update node to version 12.18.3. Its required by electron 11.
update docs with minimum version of node.
update minimum version of node in project.
update node-pty to 0.10.0. It's required by electron-rebuild
fix type error

Closes #9095.

Signed-off-by: 翰宇 林 <305991071@qq.com>



#### What it does

update electron to version 11
update electron-store to version 7. Its required by electron 11.
update node to version 12.18.3. Its required by electron 11.
update docs with minimum version of node.
update minimum version of node in project.
update node-pty to 0.10.0. It's required by electron-rebuild
fix compile error

Closes #9095.

#### How to test

  Pull changes from repository branch.
    Install node 12.18.3
    Install dependencies by running command yarn in project root directory.
    Run theia electron example by running command yarn start:electron
    In case of errors caused by native modules run command yarn
    rebuild:clean && yarn rebuild:electron and repeat previous step.
    Open chrome dev tools execute process.versions and verify that electron
    version is 11.
    Manually test functionalities.

    Use electron-builder, electron-rebuild to rebuild all native dependency and checkout those native module functionalities
#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
